testsuite/tools/expect_test.ml

```diff
@@ -237,7 +237,7 @@ let eval_expect_file _fname ~file_contents =
           exec_phrase ppf phrase
         with exn ->
           let bt = Printexc.get_raw_backtrace () in
-          begin try Location.report_exception ppf exn
+          begin try Location.report_exception (Misc.Log.Direct ppf) exn
```

Comment:
There are a lot of changes to existing calls to `Location.report_exception` here. If I understand correctly, the main compiler codebase should use a structured log, but external callers keep using the direct way. Maybe we should consider having two separate functions, `report_exception` unchanged and then `log_exception` for log, to minimize breakage among compiler-API users.

Reply:
All functions have been split between new `log_*` functions and their backward-compatible counterparts.